protocols/request-response/tests/ping.rs

```diff
@@ -64,22 +64,19 @@ fn ping_protocol() {
     let expected_pong = pong.clone();
 
     let peer1 = async move {
-        while let Some(_) = swarm1.next().now_or_never() {}
-
-        let l = Swarm::listeners(&swarm1).next().unwrap();
-        tx.send(l.clone()).await.unwrap();
-
         loop {
-            match swarm1.next().await {
-                RequestResponseEvent::Message {
+            match swarm1.next_event().await {
+                SwarmEvent::NewListenAddr(addr) => tx.send(addr).await.unwrap(),
```

Comment:
since we don't use get-if-addrs which I think blocks the thread, it's not guaranteed that the event is emitted immediately